### DISCUSSION TITLE
How to run healthcheck if we are running keycloak in container

### DISCUSSION THREAD

**Discussion by umeshauv** [keycloak/keycloak#17319] [Q&A]
We are running keycloak 21.0.0 as a container.  We want to use health check end points as docker healthcheck.  We hit an issue because the keycloak image doesn't contain curl and we can't install curl because keycloak image doesn't contain any package manager tools like dnf or yum.

**Comment by Pilpin** [comment]
Same problem here to download an identity provider.

@umeshauv would you consider changing the title of the discussion to something about package managers or curl ?

**Comment by Pilpin** [comment]
A closed issue on this subject : https://github.com/keycloak/keycloak/issues/17384
A PR for the docs : https://github.com/keycloak/keycloak/pull/17317/files

**Comment by darius-m** [comment]
A rather unwieldy option would be using something like [this stackexchange answer](https://unix.stackexchange.com/a/525681) suggests - for example, you can use `echo > /dev/tcp/localhost/8080` to see if the `8080` port has been opened. This will, of course, not work (easily) with SSL encrypted connections, but you may still be able to confirm that the port is open. For plain HTTP you can confirm that it returns an answer (you will need to use the redirect trick) described in the linked answer.

However, the docs suggested by @Pilpin are the more sensible solution.
